File: adminportal/src/Dto/AdminPortal/PostInfo.ts
import type { ThreadBasicInfo } from "./ThreadInfo"
import type { TopicBasicInfo } from "./TopicInfo"
import type { UserBasicInfo } from "./UserInfo"

export type PostEntry = {
    postId: string,
    threadId: string,
    isFirstPost: boolean,
    createdByUserId: string,
    postContent: string,
    createdDate: Date,
    replyToPostId: string,
    postReported: boolean,
    reportReason: string,
    reportedByUserId: string,
}

export type PostBasicInfo = {
    post: PostEntry,
    createdByUser: UserBasicInfo,
    thread?: ThreadBasicInfo
}

export type PostSummary = {
    totalPosts: number
}

File: server/RestApiServer.Endpoints/Dto/Forum/PostInfo.cs
using RestApiServer.Db;
using RestApiServer.Dto.App;

namespace RestApiServer.Dto.Forum
{
    public class PostBasicInfo
    {
        public required PostEntry Post { get; set; }
        public required UserBasicInfo CreatedByUser { get; set; }
    }

    public class PostFullInfo
    {
        public required PostEntry Post { get; set; }
        public required UserBasicInfo CreatedByUser { get; set; }
    }

    public class PostSummary 
    {
        public required int TotalPosts { get; set; } = 0;
    }
}